### PR TITLE
[SFA-58] [SFA-73] Rating System UI + Mock Data

### PR DESCRIPTION
**Pull request by akassab**
# Preface
- Builds off of the notification created from PR #72

# This PR
- Created `item_page.xml`
image

- Added something called a `ViewPager2` which allows a user to swipe through fragments
- Also added `Circle Indicator` which is just a UI feature that displays which position we are in the fragment 
image
- Uses a `ViewPagerAdapter` with dummy data
- The ViewPagerAdatper contains a mutable lift of floats
- Currently uses dummy data (generates 5 dummy data)

# Gif
- Note: In this GIF the notification does not appear at the same time as the `match.endTime` because I hard-coded it to trigger instantly as testing this feature can be really time-consuming. Don't worry though, I did not commit this small "hack"
ratinggif

# Future PR
- Expect a future PR that uses real players from the match instead of dummy data
- Need to add a "rating" attribute to the User model in Firestore

### CODE REVIEW

File: app/src/main/java/com/uwcs446/socialsports/ui/matchrating/MatchRatingActivity.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import android.os.Bundle
+import android.util.Log
+import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import androidx.viewpager2.widget.ViewPager2
+import com.uwcs446.socialsports.R
+import kotlinx.android.synthetic.main.activity_match_rating.*
+import me.relex.circleindicator.CircleIndicator3
+
+class MatchRatingActivity : AppCompatActivity() {
```
Comment (aaronchlam) [inline app/src/main/java/com/uwcs446/socialsports/ui/matchrating/MatchRatingActivity.kt:12]: how come match rating is an activity? Why doesn't this follow how we use fragment for everything else?

Comment (akassab) [reply app/src/main/java/com/uwcs446/socialsports/ui/matchrating/MatchRatingActivity.kt:12]: I understand that we have fragments but I don't think there is anything wrong with using an activity alongside fragments. A famous application does this (ex: Uber + Uber Eats after you finish your ride / get your food, a new activity is shown to rate).

Comment (akassab) [reply app/src/main/java/com/uwcs446/socialsports/ui/matchrating/MatchRatingActivity.kt:12]: Besides just copying whatever uber does, it also makes sense from a UX standpoint, that if the user simply received a notification to rate ... then why would they need to see the bottom navigation bar again?

Comment (aaronchlam) [reply app/src/main/java/com/uwcs446/socialsports/ui/matchrating/MatchRatingActivity.kt:12]: Our assignments talk about using fragments. I do not want this inconsistency to impact our grades. Are you not able to implement this in fragments?

Comment (akassab) [reply app/src/main/java/com/uwcs446/socialsports/ui/matchrating/MatchRatingActivity.kt:12]: Can you specify where exactly we mentioned we would use fragments for all UI? I looked over D6 and that is never explicitly mentioned.

I looked it over and there is no mentioned of fragments and the rating UI.
image

Also, given we have to present at 12pm and rating was one of the only things we have to mitigate harm is this really worth blocking at the last minute?

Comment (aaronchlam) [reply app/src/main/java/com/uwcs446/socialsports/ui/matchrating/MatchRatingActivity.kt:12]: In the User Interface subsection of the System Design section, we discuss how we use the MVVM model and it's implied that all views follow the Fragment, ViewModel and Repository pattern. We go through an example that discusses how Fragments with with layouts and ViewModels.

Converting from Activity to Fragment is fairly straightforward. I've already done so with MatchDetails when you implemented it with Activity and it was pointed out that it's not consistent with the rest of the app architecture.

Comment (akassab) [reply app/src/main/java/com/uwcs446/socialsports/ui/matchrating/MatchRatingActivity.kt:12]: The activity follows the mvvc pattern. The next branch implements the ViewModel for this activity

It's also not implied that we only use fragments as we clearly mentioned "activity" or "fragments"
image

**Review by cyamonide — APPROVED** [review]

**Pull request merged**